spec/lib/miq_expression_spec.rb

```diff
@@ -498,23 +498,48 @@
       expect(sql).to be_nil
     end
 
+    it "can't generate the SQL for a CONTAINS expression with belongs_to field" do
+      sql, * = MiqExpression.new("CONTAINS" => {"field" => "Vm.host-name", "value" => "foo"}).to_sql
+      expect(sql).to be_nil
+    end
+
+    it "can't generate the SQL for multi level contains with a scope" do
+      sql, _ = MiqExpression.new("CONTAINS" => {"field" => "ExtManagementSystem.clustered_hosts.operating_system-name", "value" => "RHEL"}).to_sql
+      expect(sql).to be_nil
+    end
+
+    it "can't generate the SQL for field belongs to 'has_and_belongs_to_many' association" do
+      sql, _ = MiqExpression.new("CONTAINS" => {"field" => "ManageIQ::Providers::InfraManager::Vm.storages-name", "value" => "abc"}).to_sql
+      expect(sql).to be_nil
+    end
+
     it "can't generate the SQL for a CONTAINS expression virtualassociation" do
       sql, * = MiqExpression.new("CONTAINS" => {"field" => "Vm.processes-name", "value" => "foo"}).to_sql
       expect(sql).to be_nil
     end
 
-    it "can't generate the SQL for a CONTAINS expression with [association.virtualassociation]" do
+    it "can't generate the SQL for a CONTAINS expression with invalid associations" do
```

Comment:
Why the change here to something less specific?

Reply:
in truth this case was passing an invalid association. had me totally confused working through it.

So I fixed this one and added what I think they meant to test a little below.

Reply:
I moved it. hope the diff is more clear